src/kubernetes/debounce.rs

```diff
@@ -0,0 +1,245 @@
+//! Arbitrary signal debouncing logic.
+//!
+//! Call [`Debounce::signal`] multiple times within the debounce time window,
+//! and the [`Debounce::debounced`] will be resolved only once.
+
+use std::time::Duration;
+use tokio::time::{delay_until, Instant};
+
+/// Provides an arbitrary signal debouncing.
+pub struct Debounce {
+    sequence_start: Option<Instant>,
+    time: Duration,
+}
+
+impl Debounce {
+    /// Create a new [`Debounce`].
+    pub fn new(time: Duration) -> Self {
+        Self {
+            sequence_start: None,
+            time,
+        }
+    }
+
+    /// Trigger a signal to debounce.
+    pub fn signal(&mut self) {
```

Comment:
That's a fair point.

But then, if the code ended up that way a - maybe we should keep it like that? To me, having it check `is_some` is more expressive. As if it decouples two concerns - "if sequence_start is already sent - do nothing" and "update sequence_start". Does it make sense?

Reply:
It does make sense, but the "do nothing" `return` that's an excess , when it could be :
"Set `sequence_start` if it isn't set"
For the two concerns, yes there are two, the condition and the action, but execution of the action is dependens on the condition so they are inherently coupled.

Reply:
Good point of view, I'll change the code as you suggested!